Packages/Blockchain/Sources/Ethereum/EthereumFeeService.swift

```diff
@@ -114,20 +115,104 @@ extension EthereumService: ChainFeeCalculateable {
         }
     }
 
-    public func feeRates() async throws -> [FeeRate] { fatalError("not implemented") }
+    public func feeRates() async throws -> [FeeRate] {
+        let (baseFee, priorityFees) = try await getBasePriorityFees(
+            blocks: Self.historyBlocks,
+            config: GemstoneConfig.shared.config(for: chain)
+        )
+        var rates: [FeeRate] = []
+        for priority in FeePriority.allCases {
+            if let priorityFee = priorityFees[priority] {
+                let feeRate = FeeRate(
+                    priority: priority,
+                    gasPriceType: .eip1559(
+                        gasPrice: baseFee,
+                        minerFee: priorityFee
+                    )
+                )
+                rates.append(feeRate)
+            }
+        }
+        return rates
+    }
+
+    func getBasePriorityFees(blocks: Int, config: EvmChainConfig) async throws -> (base: BigInt, priority: [FeePriority: BigInt]) {
```

Comment:
`config: EvmChainConfig` just pass values that you need, no need to pass `EvmChainConfig`